g2p/cli.py

```diff
@@ -531,39 +539,54 @@ def convert(  # noqa: C901
         raise click.UsageError(
             f"Path between '{in_lang}' and '{out_lang}' does not exist"
         )
-    # Figure if the input text is on the command line or in a file
-    input_text_is_a_file = os.path.exists(input_text) and input_text.endswith("txt")
-    if input_text_is_a_file:
-        with open(input_text, encoding="utf8") as f:
-            lines = f.readlines()
-    else:
-        lines = [input_text]
-    # Determine which tokenizer to use, if any
-    if tok is not None and not tok and tok_lang is not None:
-        raise click.UsageError("Specified conflicting --no-tok and --tok-lang options.")
-    if tok is None:
-        tok = True  # Tokenize by default
-    custom_tokenizer = make_tokenizer(tok_lang) if tok_lang else None
-    # Transduce!!!
-    assert in_lang and out_lang
-    transducer = make_g2p(
-        in_lang, out_lang, tokenize=tok, custom_tokenizer=custom_tokenizer
-    )
-    for line in lines:
-        tg = transducer(line)
-        if check:
-            transducer.check(tg, display_warnings=True)
-        outputs = [tg.output_string]
-        if substring_alignments:
-            outputs += [tg.substring_alignments()]
-        if pretty_edges:
-            outputs += [tg.pretty_edges()]
-        if debugger:
-            outputs += [tg.edges, tg.debugger]
-        if len(outputs) > 1:
-            click.echo(pprint.pformat(outputs, indent=4))
+    to_close = None
+    try:
+        if file:
+            if input_text == "-":
+                lines = sys.stdin
+            else:
+                try:
+                    to_close = lines = open(input_text, encoding="utf8")
```

Comment:
why not use `with open(input_text)`? just to prevent the indenting? I prefer using `with` just in case something happens later that prevents `to_close` from actually being closed.

Reply:
Because if the input is a text string, you're not going to be inside that `with` block.

Reply:
I always use `with`, it has to be impossible to use it before I resolve to using something else.